### PR TITLE
Enhancement: More information about S3 keys searched for and Dockerfile that uses submodule instead of channel

### PR DESCRIPTION
**Pull request by tzaffi**
# More info in `e2elive.py` + `Dockerfile` update

## Summary of changes
* [Improved message](https://app.circleci.com/pipelines/github/algorand/indexer/1919/workflows/2c21d2cd-c170-43da-89b8-8db814b0d33b/jobs/3124/parallel-runs/0/steps/0-122) at the end of `make e2e` (see image below)
* `misc/Dockerfile` -  ([cherry picked](https://github.com/algorand/indexer/pull/1002/commits/9d39e43a3bbe7224ce2caecf3cd3e5dfba25d736) out of @Eric-Warehime 's #1002 ) `make e2e` calls out to `docker-compose` which builds and runs this `Dockerfile` to run the end to end tests via `misc/e2elive.py` running inside the container.  In doing so, an `algod` with private network is brought up with `goal nework start -r ${tempnet}`.
  * Previously, the `go-algorand` was retrieved and built from the nightly channel via a [pinned version](https://github.com/algorand/go-algorand/raw/1e1474216421da27008726c44ebe0a5ba2fb6a08/cmd/updater/update.sh) of `go-algorand`'s`cmd/updater/update.sh` and the command `./update.sh -i -c nightly -p /opt/algorand/bin -d /opt/algorand/data -n`. 
  * Going forward, `go-algorand` is being built directly with `make install` under the submodule `third_party/go-algorand`
  * **TLDR;** - we'll be using the commit's go-algorand submodule in end to end tests instead of the external nightly build

### Example Output for `e2elive.py` Failing Due to Missing Tarball
image

## Test Plan

CI

### CODE REVIEW

**Comment by brianolson** [comment]
It's hard to read for the actual change with the formatter applied. Could we split this into a format-PR and a fix-PR?

**Comment by tzaffi** [comment]
> It's hard to read for the actual change with the formatter applied. Could we split this into a format-PR and a fix-PR?

Sure thing. So look out for a pure formatting PR coming soon.

File: misc/util.py
Comment (brianolson) [inline misc/util.py:178]: If this line was the only actual change, it would be trivial to approve.

Comment (tzaffi) [reply misc/util.py:178]: There's a bit more, but only in that function.

File: misc/e2elive.py
Comment (tzaffi) [inline misc/e2elive.py:80]: ^^^ DO NOT MERGE AS IS ^^^

**Review by Eric-Warehime — APPROVED** [review]

File: misc/e2elive.py
Comment (tzaffi) [inline misc/e2elive.py:86]:
```suggestion
        prefix = "indexer/e2e4"
        success = firstFromS3Prefix(
            s3, bucket, prefix, tarname, outpath=tarpath
        )
        if not success:
            raise Exception(
                f"failed to locate tarname={tarname} from AWS S3 path {bucket}/{prefix}"
            )
```

File: misc/util.py
Comment (brianolson) [inline misc/util.py:168]: this has the flavor I'd expect from a `--verbose` level debug print. Do we really want it on all the time?

**Review by winder — APPROVED** [review]

File: misc/util.py
Comment (tzaffi) [reply misc/util.py:168]: Didn't spot this comment in time before merging. I would advocate for it (not so much the silly Hello, but more the print out of the haystack being searched). I find it useful to see which S3 objects actually were being considered. I'm open to further discussion.

**Pull request merged**
